packages/react-charting/src/components/TreeChart/TreeChart.base.tsx

```diff
@@ -0,0 +1,293 @@
+/* eslint-disable dot-notation */
+/* eslint-disable one-var */
+/* eslint-disable @fluentui/max-len */
+/* eslint-disable prefer-arrow-callback */
+/* eslint-disable react/self-closing-comp */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable no-console */
+/* eslint-disable prefer-const */
+import * as React from 'react';
+import { createRef } from 'react';
+import { select, Selection } from 'd3-selection';
+import { hierarchy, tree } from 'd3-hierarchy';
+
+import { ITreeProps, ITreeState, ITreeDataStructure, IDataStructure } from '../../index';
+
+// Create a parent class for common tree components
+class StandardTree {
+  public treeData: IDataStructure;
+  public svg: Selection<SVGSVGElement | null, unknown, null, undefined>;
+  constructor(treeData: IDataStructure, svg: Selection<SVGSVGElement | null, unknown, null, undefined>) {
+    this.treeData = treeData;
+    this.svg = svg;
+  }
+
+  public addNodeShapetoSVG(
+    name: string,
+    subname: string,
+    xCoordinate: number,
+    yCoordinate: number,
+    fillColor: string,
+    rectangleWidth: number,
+    rectangleHeight: number,
+  ) {
+    this.svg
+      .append('rect')
+      .attr('width', rectangleWidth)
+      .attr('height', rectangleHeight)
+      .attr('x', xCoordinate)
+      .attr('y', yCoordinate)
+      .attr('padding', '10px')
+      .attr('rx', '1')
+      .style('stroke', fillColor)
+      .style('stroke-width', '2px')
+      .style('fill', 'white');
+
+    this.svg
+      .append('text')
+      .style('fill', 'black')
+      .attr('dy', yCoordinate + rectangleHeight / 2.5)
+      .attr('x', () => {
+        return xCoordinate + rectangleWidth / 3.6;
+      })
+      .attr('text-anchor', () => {
+        return 'start';
+      })
+      .text(function () {
+        return name;
+      })
+      .append('tspan')
+      .attr('dy', '1.4em')
+      .attr('x', () => {
+        return xCoordinate + rectangleWidth / 3.4;
+      })
+      .text(() => {
+        return subname;
+      });
+  }
+  // Creates a rectangular path from parent to the child nodes
+  public addLinktoNodes(child: any, parent: any, leaf: boolean, rectWidth: number, gap: number): any {
+    let path = `M${child.x + rectWidth / 2},${child.y - gap} H${parent.x + rectWidth / 2} V${parent.y}`;
+    let leafpath = `M${parent.x + rectWidth / 2},${parent.y} V${parent.y + gap * 5} H${parent.x - gap / 2} H${
```

Comment:
Add a comment on what does gap represent.

Reply:
Added these comments in the function: 
* // gap adds ratio for parent.y to child.y
* // based on the type of node return leaf-path or path element